### ISSUE TITLE
Travis CI builds of master always report success

### ISSUE THREAD

**Issue by mpharrigan** [omnia-md/conda-recipes#690]
Even when packages fail to build, it reports success because the script is
```bash
conda-build-all ... || true
```

Why are we doing this?

**Comment by jchodera** [comment]
We definitely want to do that on `master` since otherwise a broken package will mean that the build will *stop* and not build any subsequent packages.

However, on PRs, we don't want to do that---we want the build to fail if there are broken packages.

**Comment by mpharrigan** [comment]
> the build will stop and not build any subsequent packages.

I don't think that's quite right. The `|| true` isn't inside a loop. It just modifies the exit code of conda-build-all. If conda-build-all exits early (on first failed package) the `|| true` isn't going to help

https://github.com/omnia-md/conda-recipes/blob/master/devtools/docker-build.sh#L17

The right place to implement this logic is in the package building loop *inside* conda-build-all.

**Comment by mpharrigan** [comment]
Note that this logic *is* already implemented in conda-build-all. The exit code is set to the number of failed packages

**Comment by jchodera** [comment]
> The right place to implement this logic is in the package building loop inside conda-build-all.

Ah! Agreed!

**Comment by jchodera** [comment]
> Note that this logic is already implemented in conda-build-all. The exit code is set to the number of failed packages

OK, let's remove this now.

**Comment by jchodera** [comment]
The `nightli.es` appear to have revealed a new issue now:
```
Fetching package metadata ....
WARNING: The remote server could not find the noarch directory for the
requested channel with url: file:///anaconda/conda-bld/noarch
It is possible you have given conda an invalid channel. Please double-check
your conda configuration using `conda config --show`.
If the requested url is in fact a valid conda channel, please request that the
channel administrator create `noarch/repodata.json` and associated
`noarch/repodata.json.bz2` files, even if `noarch/repodata.json` is empty.
Traceback (most recent call last):
  File "/io/conda-build-all", line 460, in <module>
    sys.exit(main())
  File "/io/conda-build-all", line 120, in main
    return execute(args, p)
  File "/io/conda-build-all", line 430, in execute
    to_build = list(required_builds(metadatas, pythons, numpys, channel_urls, verbose=args.verbose, force=args.force))
  File "/io/conda-build-all", line 310, in required_builds
    prepend=False)
  File "/anaconda/lib/python3.5/site-packages/conda/core/index.py", line 124, in get_index
    index = fetch_index(channel_priority_map, use_cache=use_cache)
  File "/anaconda/lib/python3.5/site-packages/conda/core/index.py", line 532, in fetch_index
    repodatas = _collect_repodatas(use_cache, tasks)
  File "/anaconda/lib/python3.5/site-packages/conda/core/index.py", line 521, in _collect_repodatas
    repodatas = _collect_repodatas_serial(use_cache, tasks)
  File "/anaconda/lib/python3.5/site-packages/conda/core/index.py", line 494, in _collect_repodatas_serial
    for url, schan, pri in tasks]
  File "/anaconda/lib/python3.5/site-packages/conda/core/index.py", line 494, in <listcomp>
    for url, schan, pri in tasks]
  File "/anaconda/lib/python3.5/site-packages/conda/core/index.py", line 143, in func
    res = f(*args, **kwargs)
  File "/anaconda/lib/python3.5/site-packages/conda/core/index.py", line 483, in fetch_repodata
    process_repodata(repodata, url, schannel, priority)
  File "/anaconda/lib/python3.5/site-packages/conda/core/index.py", line 410, in process_repodata
    opackages = repodata.setdefault('packages', {})
AttributeError: 'NoneType' object has no attribute 'setdefault'
```

**Comment by jchodera** [comment]
Seems to be related: https://github.com/conda/conda/issues/4513

**Comment by jchodera** [comment]
Looks like a fix is in the works: https://github.com/conda/conda/pull/4518

I'm going to try to pin `conda` to a known working version to see if that fixes things.

**Comment by jlmaccal** [comment]
Should have checked here first before trying to tack this down...

I think we're hitting this because we're adding the `conda` build directory as a local channel when it no longer has the correct structure (it's missing the `noarch` directory and `repodata.json`). We should probably fix this.

Does anyone know why we're adding the build directory like this? Can we just remove it?

**Comment by Lnaden** [comment]
This was fixed by #691
